.github/workflows/e2e-tests.yml

```diff
@@ -38,9 +39,79 @@ jobs:
           token: ${{ github.token }}
           filters: .github/file-paths.yaml
 
-  build:
+  download_uberjar:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
     needs: files-changed
-    if: github.event.pull_request.draft == false && needs.files-changed.outputs.e2e_all == 'true'
+    if: |
+      !cancelled() &&
+      github.event.pull_request.draft == false &&
+      needs.files-changed.outputs.e2e_specs == 'true'
+    strategy:
+      matrix:
+        edition: [oss, ee]
+    outputs:
+      enterprise_download_link: ${{ steps.zip_url.outputs.ee_dl }}
+      oss_download_link: ${{ steps.zip_url.outputs.oss_dl }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Artifact zip URL
+        id: zip_url
+        run: |
+          current_commit="HEAD"
+
+          get_parent () {
+            parent_commit=$(git rev-parse $1^)
+            echo $parent_commit
+          }
+
+          get_download_link () {
+            parent_commit=$(get_parent $1)
+            current_page=${2:-1}
```

Comment:
Optional second argument with the default value
https://stackoverflow.com/a/9333006/8815185